pkg/types/defaults/installconfig.go

```diff
@@ -1,6 +1,11 @@
 package defaults
```

Comment:
I'm not sure if package `defaults` is the right place for this. I think of `defaults` as "user didn't supply a value, so here's the default" but that's not what's happening here (users are supplying a value). 

Userssupply a pull secret when they fill out the survey to generate an install config. That code is here: https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/pullsecret.go So it would seem possible to load the pull secret from the flag/env var and skip this survey step if the pull secret is supplied.

The other way users supply pull secrets is through the install config, so potentially we could make the pull secret field optional and then validate that either the install config contains the pull secret or that the envvar/flag is set. 

I'm not sure if there is a third path for the ABI to get the pull secret?

Reply:
ABI has an override meant to support [ZTP manifests](https://github.com/openshift/installer/blob/84d8873e9df243d067f5ada0f9e43eaaf97aaba3/pkg/asset/agent/manifests/agentpullsecret.go#L84), otherwise it follows the usual path of retrieving it from the [install-config](https://github.com/openshift/installer/blob/84d8873e9df243d067f5ada0f9e43eaaf97aaba3/pkg/asset/agent/manifests/agentpullsecret.go#L64).

Anyhow, since for the agent flow the whole install-config may considered optional (in case the user provided all the necessary ZTP manifests), we had a special asset called [OptionalInstallConfig](https://github.com/openshift/installer/blob/84d8873e9df243d067f5ada0f9e43eaaf97aaba3/pkg/asset/agent/installconfig.go#L55): if the user provided the install-config.yaml file, then it's used. The main issue here is that the original install-config `Generate` is never exercised  (while the [defaults](https://github.com/openshift/installer/blob/84d8873e9df243d067f5ada0f9e43eaaf97aaba3/pkg/asset/installconfig/installconfigbase.go#L68) are, thanks to the original loading).